src/main/java/io/debezium/connector/vitess/connection/VStreamOutputMessageDecoder.java

```diff
@@ -388,4 +388,25 @@ else if (uniqueKeyColumnName != null) {
 
         return tableEditor.create();
     }
+
+    private static String validateColumnName(String columnName, String schemaName, String tableName) {
```

Comment:
All this validation logic applied only when we serialize to AVRO format. I guess it is not the case when JSON is used to serialize events...

Reply:
As discussed offline, Vitess VStreamer's schema reloading [bug](https://vitess.slack.com/archives/C0PQY0PTK/p1607500147157900?thread_ts=1606817216.038500&cid=C0PQY0PTK) can cause the column name in the event to have `@` prefix. In this case, I think it’s better to fail than to send the corrupted row event with `@1` `@2` as column names.

Reply:
Can we distinguish those really invalid column names from ones which just aren't Avro-compatible? For the latter, all other connectors support the [sanitize.field.names](https://debezium.io/documentation/reference/connectors/postgresql.html#postgresql-property-sanitize-field-names) option, which will replace any invalid character when using Avro. Ideally, we'd have the same option and behavior for Vitess, too. Then failing hard only on those bug-caused names would be ideal, I think.

Reply:
@gunnarmorling thanks for the suggestion. I'll double-check if Vitess Connector supports the [sanitize.field.names](sanitize.field.names) option. If not supported, I'll create a ticket to add the support for it.

But indeed, for this specific case, it is a Vitess schema reloading bug that causes the normal column names (e.g. `foo`, `bar`) to be anonymized (e.g. `@1`, `@2`).

Reply:
I'll also document this behavior in the Vitess Documentation once this PR is approved.